osm2gtfs/creators/trips_creator.py

```diff
@@ -13,32 +19,301 @@ def __repr__(self):
 
-    @staticmethod
-    def add_shape(feed, route_id, osm_r):
+        # Check if itinerary and line are having the same reference
+        if itinerary.route_id != line.route_id:
+            print("Warning: The route id of the itinerary (" +
+                  str(itinerary.route_id) + ") doesn't match route if of line (" +
+                  str(line.route_id) + ")")
+            print(" " + itinerary.osm_url)
+            print(" " + line.osm_url)
+            return True
+
+        # Check if time information in schedule can be found for
+        # the itinerary
+        if itinerary.route_id not in schedule['lines']:
+            print(" Warning: Route not found in schedule.")
+            return False
+
+        return True
+
+    def _add_shape_to_feed(self, feed, shape_id, itinerary):
         """
-        create GTFS shape and return shape_id to add on GTFS trip
+        Create GTFS shape and return shape_id to add on GTFS trip
         """
-        import transitfeed
-        shape_id = str(route_id)
+        shape_id = str(shape_id)
+
+        # Only add a shape if there isn't one with this shape_id
         try:
             feed.GetShape(shape_id)
         except KeyError:
             shape = transitfeed.Shape(shape_id)
-            for point in osm_r.shape:
+            for point in itinerary.shape:
                 shape.AddPoint(
                     lat=float(point["lat"]), lon=float(point["lon"]))
             feed.AddShapeObject(shape)
         return shape_id
+
+    def _add_itinerary_trips(self, feed, itinerary, line, trip_builder,
+                             shape_id):
+        """
+        Add all trips of an itinerary to the GTFS feed.
+        """
+        # Obtain GTFS route to add trips to it.
+        route = feed.GetRoute(line.route_id)
+        trips_count = 0
+
+        # Loop through each timeslot for a trip
+        for trip in trip_builder['schedule']:
+            gtfs_trip = route.AddTrip(feed, headsign=itinerary.to,
+                                      service_period=trip_builder['service_period'])
+            trips_count += 1
+
+            # Go through all stops of an itinerary
+            for itinerary_stop_id in itinerary.get_stops():
+
+                # Load full stop object
+                try:
+                    itinerary_stop = trip_builder[
+                        'all_stops']['regular'][itinerary_stop_id]
+                except ValueError:
+                    print("Itinerary (" + itinerary.route_url +
+                          ") misses a stop:")
+                    print(" Please review:" + itinerary_stop.osm_url)
+                    continue
+
+                try:
+                    # Load respective GTFS stop object
+                    gtfs_stop = feed.GetStop(str(itinerary_stop.stop_id))
+                except ValueError:
+                    print("Warning: Stop in itinerary was not found in GTFS.")
+                    print(itinerary_stop.osm_url)
```

Comment:
Add an explication to print :)

Reply:
What do you mean? An explication to print?

Reply:
I wanted to say something like "OSM-Stop Url:"